### PR TITLE
OB-1: Sum functionality added

### PR DESCRIPTION
**Pull request by davidburgos**
Support for sum two integer values.
Testing Done: Sony Xperia Z5
Signed-off-by: David Burgos david.burgos@globant.com

### CODE REVIEW

File: app/src/main/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenter.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    private CalculatorView view;
+
+    public CalculatorPresenter(CalculatorModel model, CalculatorView view) {
+        this.model = model;
+        this.view = view;
+    }
+
+    private void onSolveButtonPressed() {
+
+        int result = 0;
```
Comment (andresuarezz26) [inline app/src/main/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenter.java]: In the project's checkstyle if you put the 0, it is marked as a warning named "Magic numbers". Please create a constant (private static final int ZERO_INT = 0;) or you can create a class to have all the constants.

File: app/src/main/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenter.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        switch (model.getOperator()) {
+            case PLUS:
+                result = model.getFirstValue() + model.getSecondValue();
+                break;
+            case UNKNOWN:
+                result = -1;
```
Comment (andresuarezz26) [inline app/src/main/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenter.java]: Magic number

File: app/src/main/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenter.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import com.davidburgos.calculator.mvp.model.CalculatorModel;
+import com.davidburgos.calculator.mvp.model.Operators;
+import com.davidburgos.calculator.mvp.view.CalculatorView;
+import com.davidburgos.calculator.util.bus.RxBus;
+import com.davidburgos.calculator.util.bus.observers.CalculatorButtonBusObserver;
+import com.davidburgos.calculator.util.bus.observers.CalculatorFirstValueBusObserver;
+import com.davidburgos.calculator.util.bus.observers.CalculatorOperatorBusObserver;
+import com.davidburgos.calculator.util.bus.observers.CalculatorSecondValueBusObserver;
+
+public class CalculatorPresenter {
```
Comment (andresuarezz26) [inline app/src/main/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenter.java]: You must do the unit test for the Presenter

File: app/src/main/java/com/davidburgos/calculator/MainActivity.java [hunk 4]
```diff
@@ -1,15 +1,32 @@
...
 
 public class MainActivity extends AppCompatActivity {
 
+    private CalculatorPresenter presenter;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        this.presenter = new CalculatorPresenter(new CalculatorModel(), new CalculatorView(this));
```
Comment (andresuarezz26) [inline app/src/main/java/com/davidburgos/calculator/MainActivity.java]: The reserved word this is redundant

File: app/src/main/java/com/davidburgos/calculator/mvp/model/CalculatorModel.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package com.davidburgos.calculator.mvp.model;
+
+public class CalculatorModel {
+
+    private int firstValue = 0;
+    private int secondValue = 0;
```
Comment (andresuarezz26) [inline app/src/main/java/com/davidburgos/calculator/mvp/model/CalculatorModel.java]: These are magic numbers for the checkstyle. Please add constants for ZERO_INT

File: app/src/main/java/com/davidburgos/calculator/mvp/view/CalculatorView.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    @BindView(R.id.calculator_result)
+    AppCompatTextView result;
+
+    public CalculatorView(Activity activity) {
+        super(activity);
+        ButterKnife.bind(this, activity);
+    }
+
+    public void setResult(String value) {
+        this.result.setText(value);
```
Comment (andresuarezz26) [inline app/src/main/java/com/davidburgos/calculator/mvp/view/CalculatorView.java]: this is redundant. You can see when to use this in the following post: https://stackoverflow.com/questions/2411270/when-should-i-use-this-in-a-class

File: app/src/main/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenter.java (hunk 3 above)
Comment (davidburgos) [reply app/src/main/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenter.java]: Done

File: app/src/main/java/com/davidburgos/calculator/MainActivity.java (hunk 4 above)
Comment (davidburgos) [reply app/src/main/java/com/davidburgos/calculator/MainActivity.java]: Done

File: app/src/main/java/com/davidburgos/calculator/mvp/model/CalculatorModel.java (hunk 5 above)
Comment (davidburgos) [reply app/src/main/java/com/davidburgos/calculator/mvp/model/CalculatorModel.java]: Done

File: app/src/main/java/com/davidburgos/calculator/mvp/view/CalculatorView.java (hunk 6 above)
Comment (davidburgos) [reply app/src/main/java/com/davidburgos/calculator/mvp/view/CalculatorView.java]: Done

File: app/src/test/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenterTest.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        presenter.onSolveButtonPressed();
+        verify(view).showMessage(anyInt());
+        verify(view, never()).setResult(anyString());
+        verifyNoMoreInteractions(view);
+    }
+
+    @Test
+    public void shouldReturnSumOfTwoIntValues() throws Exception {
+        when(model.getOperator()).thenReturn(Operators.PLUS);
+        when(model.getFirstValue()).thenReturn(10);
```
Comment (andresuarezz26) [inline app/src/test/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenterTest.java]: Remember the magic numbers. Is a little annoying but is for not generate checkstyle warnings

File: app/src/test/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenterTest.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    public void shouldShowMessageWhenOperatorIsUnknown() throws Exception {
+        when(model.getOperator()).thenReturn(Operators.UNKNOWN);
+        presenter.onSolveButtonPressed();
+        verify(view).showMessage(anyInt());
+        verify(view, never()).setResult(anyString());
+        verifyNoMoreInteractions(view);
+    }
+
+    @Test
+    public void shouldReturnSumOfTwoIntValues() throws Exception {
```
Comment (andresuarezz26) [inline app/src/test/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenterTest.java]: The test is fine! good work

**Pull request merged**